mikrotik/resource_interface_wireguard_peer.go

```diff
@@ -0,0 +1,273 @@
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64default"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	tftypes "github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type interfaceWireguardPeer struct {
+	client *client.Mikrotik
+}
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &interfaceWireguardPeer{}
+	_ resource.ResourceWithConfigure   = &interfaceWireguardPeer{}
+	_ resource.ResourceWithImportState = &interfaceWireguardPeer{}
+)
+
+// NewInterfaceWireguardPeerResource is a helper function to simplify the provider implementation.
+func NewInterfaceWireguardPeerResource() resource.Resource {
+	return &interfaceWireguardPeer{}
+
+}
+
+func (i *interfaceWireguardPeer) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	i.client = req.ProviderData.(*client.Mikrotik)
+}
+
+// Metadata returns the resource type name.
+func (i *interfaceWireguardPeer) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_interface_wireguard_peer"
+}
+
+// Schema defines the schema for the resource.
+func (i *interfaceWireguardPeer) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Creates a Mikrotik Interface Wireguard Peer only supported by RouterOS v7+.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Description: "Identifier of this resource assigned by RouterOS",
+			},
+			"allowed_address": schema.StringAttribute{
+				Optional:    true,
+				Description: "List of IP (v4 or v6) addresses with CIDR masks from which incoming traffic for this peer is allowed and to which outgoing traffic for this peer is directed. The catch-all 0.0.0.0/0 may be specified for matching all IPv4 addresses, and ::/0 may be specified for matching all IPv6 addresses.",
+			},
+			"comment": schema.StringAttribute{
+				Optional:    true,
+				Computed:    true,
+				Description: "Short description of the peer.",
+			},
+			"disabled": schema.BoolAttribute{
+				Optional:    true,
+				Computed:    true,
+				Default:     booldefault.StaticBool(false),
+				Description: "Boolean for whether or not the interface peer is disabled.",
+			},
+			"endpoint_address": schema.StringAttribute{
+				Optional:    true,
+				Computed:    true,
```

Comment:
What was the rationale for making this computed? I think this should also should default to an empty string and not be computed. If `Computed` is removed, the default value of `""` should apply.